src/sentry/discover/models.py

```diff
@@ -8,15 +8,33 @@
 from sentry.backup.scopes import RelocationScope
 from sentry.db.models import BaseManager, FlexibleForeignKey, Model, region_silo_model, sane_repr
 from sentry.db.models.fields import JSONField
-from sentry.db.models.fields.bounded import BoundedBigIntegerField
+from sentry.db.models.fields.bounded import BoundedBigIntegerField, BoundedPositiveIntegerField
 from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+from sentry.models.dashboard_widget import TypesClass
 from sentry.models.projectteam import ProjectTeam
 from sentry.tasks.relay import schedule_invalidate_project_config
 
 MAX_KEY_TRANSACTIONS = 10
 MAX_TEAM_KEY_TRANSACTIONS = 100
 
 
+class DiscoverSavedQueryTypes(TypesClass):
+    ERROR_EVENTS = 0
+    """
+     Error side of the split from Discover.
+    """
+    TRANSACTION_LIKE = 1
```

Comment:
I'm just using the same names we do on Dashboards

Reply:
Or maybe it's better to name them same as actual datasets? I don't have any strong opinions

Reply:
Feels like it should be the real datasets? Not sure exactly how this is used though

Reply:
The problem with using actual datasets, is that in some cases (eg. dashboards) the `Transactions` side of `Errors & Transaction` is not the actual dataset / indexed events, but a series of checks and attempts against using metrics that punch past DS. 

Hence "transactions-like".

Reply:
yeah, agreed kevan. i thought about it a bit more, and it needs to be be the "dataset" that the frontend thinks we're using. we'll add trace search to the dataset selector probably, and both transactions and trace search will be hitting the spans dataset.